src/lib/LUA/lua.cpp

```diff
@@ -116,6 +116,17 @@ static uint8_t *luaInt16StructToArray(const void *luaStruct, uint8_t *next)
   return (uint8_t *)stpcpy((char *)next, p1->units);
 }
 
+/***
```

Comment:
Tied to what define? `#if 0`?

Reply:
Yes, or a ENABLE_FLOAT_SUPPORT or something, which would be defined at the top of the source file.

Reply:
Well the idea is that if we need it then it would have to be there for everyone. If it is ifdef'd then it looks really messy to me
```cpp
    case CRSF_INT16: // fallthrough
    case CRSF_UINT16:
      dataEnd = luaInt16StructToArray(luaData, chunkStart);
      break;
    //case CRSF_FLOAT:
    //  dataEnd = luaFloatStructToArray(luaData, chunkStart);
      break;
    case CRSF_STRING: // fallthough
    case CRSF_INFO:
      dataEnd = luaStringStructToArray(luaData, chunkStart);
      break;
    case CRSF_FLOAT:
    case CRSF_OUT_OF_RANGE:
    default:
      return 0;
```
would become the less-readable
```cpp
    case CRSF_INT16: // fallthrough
    case CRSF_UINT16:
      dataEnd = luaInt16StructToArray(luaData, chunkStart);
      break;
#if defined(ENABLE_CRSF_LUA_FLOAT_TYPE)
    case CRSF_FLOAT:
      dataEnd = luaFloatStructToArray(luaData, chunkStart);
      break;
#endif
    case CRSF_STRING: // fallthough
    case CRSF_INFO:
      dataEnd = luaStringStructToArray(luaData, chunkStart);
      break;
#if !defined(ENABLE_CRSF_LUA_FLOAT_TYPE)
    case CRSF_FLOAT:
#endif
    case CRSF_OUT_OF_RANGE:
    default:
      return 0;
```

I can also remove it entirely if you think the commented out stuff is bad. It isn't complicated code so it is trivial to put it back in if we ever need it (fingers crossed we don't)

Reply:
Maybe we should just have the code there then even though it's not used? On the STM devices it will be removed by the LTO anyway (except the siyi 😄 ), and the others have plenty of space.

I agree that it looks a bit ugly with the ifdefs.

Reply:
haha `-flto` save us! But actually it won't because the compiler can't ever tell what data type the serializer will be called with so it has to keep the float case around _in case_ it comes up.

Should I just delete it and the examples? If so I'll paste the code here to the PR comments so we can copy paste it back in if we ever need it, in some fantastic future where RAM on the handset isn't a problem any more and we all live on the moon. :-D

Reply:
You could add a comment pointing to this ticket so when looking at the code (when we're on the moon) then we can just grab it from here.

Reply:
I've removed it entirely, moon-based us can go pound regolith!